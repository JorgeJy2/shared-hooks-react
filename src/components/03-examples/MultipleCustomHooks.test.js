import {
    shallow
} from 'enzyme';
import React from 'react';
import {
    useCounter
} from '../01-useState/hooks/useCounter';
import {
    useFetch
} from './hooks/useFetch';
import {
    MultipleCustomHooks
} from './MultipleCustomHooks';

jest.mock('./hooks/useFetch');
jest.mock('../01-useState/hooks/useCounter');


describe('test MultipleCustomHooks ', () => {

    test('wrapper ', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        useCounter.mockReturnValue({
            state: 1,
            increment: () => {},
            decrement: () => {},
            reset: () => {}
        });

        const wraper = shallow( < MultipleCustomHooks / > );
        expect(wraper).toMatchSnapshot();
    });

    test('should loading', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        useCounter.mockReturnValue({
            state: 1,
            increment: () => {},
            decrement: () => {},
            reset: () => {}
        });

        const wraper = shallow( < MultipleCustomHooks / > );
        const loading = wraper.find('#loading');
        expect(loading).not.toBeNull();
        expect(loading.text()).toEqual('Loading . . .');
    });

    test('should show info', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Jorge',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        });

        useCounter.mockReturnValue({
            state: 1,
            increment: () => {},
            decrement: () => {},
            reset: () => {}
        });

        const wraper = shallow( < MultipleCustomHooks / > );

        expect(wraper.find('#loading').exists()).toBeFalsy();
        expect(wraper.find('h3').text()).toBe('Hola mundo');
        expect(wraper.find('footer').text()).toBe('Jorge');

    })

});