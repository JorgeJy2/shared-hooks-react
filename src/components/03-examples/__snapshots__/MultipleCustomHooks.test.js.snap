// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MultipleCustomHooks  wrapper  1`] = `
<Fragment>
  <h1>
    Custom hooks
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
    id="loading"
  >
    Loading . . .
  </div>
</Fragment>
`;
